File: festival-web/src/main/java/cn/bdqfork/web/VertxAware.java
package cn.bdqfork.web;

import cn.bdqfork.core.exception.BeansException;
import io.vertx.core.Vertx;

/**
 * @author bdq
 * @since 2020/1/26
 */
public interface VertxAware {
    void setVertx(Vertx vertx) throws BeansException;
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/VertxAware.kt
package cn.bdqfork.kotlin.web

import cn.bdqfork.core.exception.BeansException
import io.vertx.core.Vertx

/**
 * @author bdq
 * @since 2020/1/26
 */
interface VertxAware {
    @Throws(BeansException::class)
    fun setVertx(vertx: Vertx)
}